src/Npgsql/NpgsqlDataSource.cs

```diff
@@ -56,6 +56,8 @@ public abstract class NpgsqlDataSource : DbDataSource
 
     bool _isBootstrapped;
 
+    DatabaseStateInfo _databaseStateInfo = new();
```

Comment:
Just in case could you make it `volatile`? While it shouldn't be strictly necessary, it can help if jit will attempt to do something weird (like reload the variable multiple times).

Reply:
Sure, will do.